kernel_gateway/services/kernels/handlers.py

```diff
@@ -19,6 +20,7 @@ class MainKernelHandler(TokenAuthorizationMixin,
     """Extends the notebook main kernel handler with token auth, CORS, and
     JSON errors.
     """
+    @gen.coroutine
     def post(self):
```

Comment:
Interesting that all the tests passed and manual testing worked too. Maybe didn't matter because the `super` does the `finish()` on the connection?

At any rate, I'll update to do option 2.